packages/next/src/build/webpack/loaders/next-flight-css-loader.ts

```diff
@@ -34,12 +34,20 @@ const NextServerCSSLoader = function (this: any, content: string) {
       .substring(0, 12)
 
     if (isCSSModule) {
-      return (
-        content + '\nmodule.exports.__checksum = ' + JSON.stringify(checksum)
-      )
+      return `\
+${content}
+module.exports.__checksum = ${JSON.stringify(checksum)}
+`
     }
 
-    return `export default ${JSON.stringify(checksum)}`
+    // Server CSS imports are always available for HMR, so we attach
+    // `module.hot.accept()` to the generated module.
+    const hmrCode = 'if (module.hot) { module.hot.accept() }'
+
+    return `\
+export default ${JSON.stringify(checksum)}
```

Comment:
How is this checksum used?

By self-accepting you basically don't send a new checksum to consuming modules. Not sure which effects that has...

Reply:
It isn't actually used - we just need to track server component changes in the hot reloader, and it's done by comparing modules' checksums. However server layer CSS imports will be ignored, so we changed it to be explicitly returning a checksum of the content hence we can know that a server refresh needs to be triggered (although this has many flaws e.g. `@import`s).